packages/netsuite-adapter/src/client/deploy_xml_utils.ts

```diff
@@ -0,0 +1,93 @@
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+import xmlParser from 'fast-xml-parser'
+import osPath from 'path'
+import _ from 'lodash'
+import { logger } from '@salto-io/logging'
+import { CONFIG_FEATURES } from '../constants'
+import { Graph, SDFObjectNode } from './graph_utils'
+import { CustomizationInfo, FileCustomizationInfo, FolderCustomizationInfo } from './types'
+import { isFileCustomizationInfo, isFolderCustomizationInfo } from './utils'
+
+type FileCabinetCustomType = FileCustomizationInfo | FolderCustomizationInfo
+const log = logger(module)
+
+export const XML_FILE_SUFFIX = '.xml'
+export const FILE_CABINET_DIR = 'FileCabinet'
+export const OBJECTS_DIR = 'Objects'
+export const ATTRIBUTES_FOLDER_NAME = '.attributes'
+export const FOLDER_ATTRIBUTES_FILE_SUFFIX = `.folder.attr${XML_FILE_SUFFIX}`
+export const ATTRIBUTES_FILE_SUFFIX = `.attr${XML_FILE_SUFFIX}`
+// The '/' prefix is needed to make osPath.resolve treat '~' as the root
+const PROJECT_ROOT_TILDE_PREFIX = '/~'
+
+const isFileCabinetCustomType = (customizationInfo: CustomizationInfo): customizationInfo is FileCabinetCustomType =>
+  isFileCustomizationInfo(customizationInfo) || isFolderCustomizationInfo(customizationInfo)
+
+export const getCustomTypeInfoPath = (dirPath: string, servicId: string, fileExtension: string): string =>
+  osPath.resolve(dirPath, OBJECTS_DIR, `${servicId}${fileExtension}`)
+
+export const getFileCabinetTypesPath = (dirPath: string, fileCabinetCustTypeInfo: FileCabinetCustomType): string => {
+  if (isFileCustomizationInfo(fileCabinetCustTypeInfo)) {
+    return osPath.resolve(dirPath, 'FileCabinet', ...fileCabinetCustTypeInfo.path.slice(0, -1))
+  }
+  return osPath.resolve(dirPath, 'FileCabinet', ...fileCabinetCustTypeInfo.path)
+}
+
+const getFileOrFolderString = (fileCabinetCusTypeInfo: FileCabinetCustomType): string => {
+  const filename = fileCabinetCusTypeInfo.path.slice(-1)[0]
+  if (isFolderCustomizationInfo(fileCabinetCusTypeInfo)) {
+    return `${getFileCabinetTypesPath(PROJECT_ROOT_TILDE_PREFIX, fileCabinetCusTypeInfo)}${osPath.sep}*`
+  }
+  return `${getFileCabinetTypesPath(PROJECT_ROOT_TILDE_PREFIX, fileCabinetCusTypeInfo)}${osPath.sep}${filename}`
+}
+
+export const reorderDeployXml = (
+  deployContent: string,
+  dependencyGraph: Graph<SDFObjectNode>,
+): string => {
+  const nodesInTopologicalOrder = dependencyGraph.getTopologicalOrder()
+    .map(node => node.value)
+    .filter(node => node.customizationInfo.typeName !== CONFIG_FEATURES)
+  const [fileCabinetNodes, objectNodes] = _.partition(
+    nodesInTopologicalOrder, node => isFileCabinetCustomType(node.customizationInfo)
+  )
```

Comment:
this way you don't need the typecasting in line 83
```suggestion
  const custInfosInTopologicalOrder = dependencyGraph.getTopologicalOrder()
    .map(node => node.value.customizationInfo)
    .filter(custInfo => custInfo.typeName !== CONFIG_FEATURES)

  const [fileCabinetCustInfos, customTypeInfos] = _.partition(
    custInfosInTopologicalOrder, custInfo => isFileCabinetCustomType(custInfo)
  )
```